### ISSUE TITLE
Possible Recursive Query Bug

### ISSUE THREAD

**Issue by artburkart** [influxdata/influxdb#8665]
### Directions
I opened a discussion on the [InfluxDB Community Site](https://community.influxdata.com/t/write-failed-for-shard-field-type-conflict-service-write/1887), but no one has responded. I think my question might fall under the category of actionable, so I'll try here. Please feel free to close if it's the wrong place.

### Bug report

__System info:__ 
```
$ influx
Connected to http://localhost:8086 version 1.2.0
InfluxDB shell version: 1.2.0
```

```
$ cat /etc/*-release | head -n1
CentOS Linux release 7.2.1511 (Core)
```

__Steps to reproduce:__

```shell
# Ran this query through Grafana:
SELECT mean("thread_cache_needed") AS "Thread Cache Needed" FROM (SELECT 100 - (("threads_created" / "connections") * 100) AS "thread_cache_needed" FROM "mysql" WHERE $timeFilter AND "host" = 'ip-10-20-0-127.ec2.internal' fill(null)) WHERE $timeFilter GROUP BY time($__interval) fill(null)
```

__Expected behavior:__ I should get result.

__Actual behavior:__ The latest data shard corrupts and days of data are lost. Please see crash log:
https://gist.github.com/artburkart/bae539db2d6d85fdf4237b485ed13312

If I haven't provided enough info, please don't hesitate to ask for more. I'm not sure what all is important to help solve this problem.

**Comment by jsternberg** [comment]
This was fixed in 1.2.1. Related to #7919.

**Comment by jsternberg** [comment]
And my apologies about the forum. Thank you for reporting this. This is definitely the appropriate place to ask for help if you see something like this.
